juju/juju
# Support application config

## Description of change

Add support for managing distinct application config separate from charm config.
Broken up into a few commits:
- Add schema for application config values
- Additional config for k8s provisioning of various resources
- Add support for storing application config in state database
- Add api facade methods for new application config APIs
- Add support for juju config and deploy to support application config

Next PRs will wire up the k8s provisioning side of things and use the proper config schema.

## QA steps

bootstrap
juju deploy mysql --config juju-external-hostname=exthost --config vip=foo
juju config mysql
-> check that output has application config and charm config correctly set

juju config --reset vip,juju-external-hostname
-> check that values are reset

juju config juju-external-hostname=foo
-> check value is set
